Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM spinalhdl/dev as spinalhdl
```

Comment:
`spinalhdl/spinalhdl` and `spinalhdl/riscv` are ready-to-use images for the end-user. They have a version of SpinalHDL preinstalled, which is known to work because it is tested just before being pushed. The idea is that, for each push to this repo, the following steps are executed:

- Compile and test SpinalHDL in a `spinalhdl/dev`.
- If successful, install SpinalHDL in new image (`spinalhdl/spinalhdl`) and push it to dockerhub.
  - Add RISCV toolchain to the previous one and push it to dockerhub (`spinalhdl/riscv`).

The advantage of including the dockerfile and building the images in this repo, is that images can be tagged when a release is pushed. Otherwise, we would need some additional mechanism to trigger a build in a separate repo.

However, I am not very sure about how are dependencies managed in Scala. Is `sbt clean compile publishLocal` enough to ensure that all the required dependencies are included in the image? I.e., that the container can be used for offline development? Also, is it necessary to keep the git repo or can it be removed after `publishLocal`?

Reply:
> The advantage of including the dockerfile and building the images in this repo, is that images can be tagged when a release is pushed. Otherwise, we would need some additional mechanism to trigger a build in a separate repo.

My argumentation is based only on personal preference; I like to keep a logical separation between repo(TL;DR: I'm ok to have a dockerfile for spinalhdl, but derivative container should live in another repo). I don't see a easy way to have a separate repo and trigger a build on each release on the other (webhooks can be used, but i feel is more a hack than a solution), so I'm not totally against the inclusion of the dockerfile in the main repo.

> However, I am not very sure about how are dependencies managed in Scala. Is `sbt clean compile publishLocal` enough to ensure that all the required dependencies are included in the image? I.e., that the container can be used for offline development? Also, is it necessary to keep the git repo or can it be removed after `publishLocal`?

It should be enough to launch `sbt clean publishLocal` , and it will populate the ~/.ivy folder (is the cache that sbt use for fetching dependencies) with the necessary files. sbt will fetch the artifacts/files from the ivy cache... so you don't need the source code on the machine.